content/articles/oop-in-java-abstraction-and-encapsulation/index.md

```diff
@@ -1,350 +1,482 @@
+In the main method. The Car represents the interface class. Benz represents the concrete class or child class. The child class gives its memory to the Car object to have access to its implemented methods. The Benz class implemented the start and stop behaviors of the Car class in the manner it wanted. This makes our application robust. We can change or add requirements to our application without breaking down the application.
+For instance, we want to add another car that stops or starts differently, we can create a car class say Toyota that implements class Car. Override its stop, start methods and implements it the way we want our Toyota to start and stop. There is no need to go into the Car class to change our code whenever there are new requirements.
 
 Now, that we’ve finally gotten abstraction off the list, on to the next one, Encapsulation.
 
-
-#**ENCAPSULATION**
+# ENCAPSULATION
 
 Imagine, you are building an application and the user can directly access and manipulate your data fields, how costly would that be? The outcome is better left to imagine than experience it.
-  
-Apart from Inheritance and Polymorphism which I will write about in my next article and Abstraction, Encapsulation is another fundamental concept of OOP that every OOP developer needs to get familiar with.
 
-Encapsulation is simply the process of encapsulating and hiding the details of a class implementation from the user. Encapsulation is achieved by wrapping the class attributes or fields(variables) and the methods(i.e the codes that manipulate the attributes)together as a unit.
+Apart from Inheritance and Polymorphism which I will write about in my next article. Encapsulation is another fundamental concept of OOP that every developer needs to get familiar with.
+
+Encapsulation is simply the process of encapsulating and hiding the details of a class implementation from the user. Encapsulation is achieved by wrapping the class attributes or variables and the methods together as a unit.
 
 Encapsulation makes for the security of an application as the user of a code cannot access the private variables of an encapsulated class except through its public methods which in most cases are the accessor and mutator methods.
```

Comment:
Please give further background on what accessor and mutator methods are. It doesn't have to be long. I would say one sentence for accessor methods, and another for mutator methods.